src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -186,14 +188,29 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
+                        {
+                            length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                            if (length == chars.Length)
+                            {
+                                char[] toReturn = chars;
+                                chars = null;
+                                ArrayPool<char>.Shared.Return(toReturn);
+                                minimumLength = Math.Min(minimumLength * 2, short.MaxValue);
```

Comment:
We grow immediately from 1K to 32K?

Reply:
Do you mean 260 -> 520 -> 1040 -> 32K?

Reply:
I misread the lower bound, so it would actually be 512b (which is what trying to rent 260b will give back), but the same question applies.  Why are we using Math.Min here, and why with short.MaxValue?

Reply:
Now I agree with @jkotas on https://github.com/dotnet/runtime/pull/33505#discussion_r391792059 since `ArrayPool<T>.Rent` returns arrays with aligned sizes so we'll just end up with `short.MaxValue` + 1, which seems to be the maximum path length.

`Math.Min` was used to prevent over-allocation when the new length was calculated based on `minimumLength` so we would iterating with 260 -> 520 -> 1040 -> 2080 -> 4160 -> 8320 -> 16640 -> ~33280~32768. @stephentoub, did you misread `Math.Min` as `Math.Max`...?

Reply:
Ugh, yes.